### PR TITLE
Split out mongoex connection from load so system is not required

### PR DESCRIPTION
**Pull request by SemanticallyNull**

### CODE REVIEW

File: mongoex/mongo.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	}
+
+	if cfg.UseTLS {
+		opts = opts.SetTLSConfig(&tls.Config{
+			MinVersion: tls.VersionTLS12,
+			RootCAs:    rootcerts.ServerCertPool(),
+		})
+	}
+
+	client, err = mongo.Connect(ctx, opts)
```
Comment (pete-woods) [inline mongoex/mongo.go:52]: Can just return this line

Comment (SemanticallyNull) [reply mongoex/mongo.go:52]: Does that not stop the defer error?

File: mongoex/mongo_test.go [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+)
+
+func TestMongoConnection(t *testing.T) {
+	ctx := testcontext.Background()
+	cfg := Config{
+		URI:    "mongodb://root:password@localhost:27107",
+		UseTLS: false,
+	}
+
+	client, err := New(ctx, "connection-test", cfg)
```
Comment (pete-woods) [inline mongoex/mongo_test.go:19]: We should close this client in a defer

Comment (SemanticallyNull) [reply mongoex/mongo_test.go:19]: Isn't that the same client we're using to return the DB?

File: mongoex/mongo.go (hunk 1 above)
Comment (pete-woods) [reply mongoex/mongo.go:52]: It does not, because it is a named return.

File: mongoex/mongo_test.go (hunk 2 above)
Comment (SemanticallyNull) [reply mongoex/mongo_test.go:19]: The disconnect does that...

Comment (SemanticallyNull) [reply mongoex/mongo_test.go:19]: Nope, I need to look at filenames better. Sorry 🤦‍♀️

**Review by pete-woods — APPROVED** [review]

**Pull request merged**